flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -365,13 +364,6 @@ boolean isWaitingForFloatingBuffers() {
 	 */
 	@Override
 	public boolean notifyBufferAvailable(Buffer buffer) {
-		// Check the isReleased state outside synchronized block first to avoid
```

Comment:
I do not understand what `deadlock`/situation was this comment mentioning and that makes me uncomfortable with removing it. Do you understand what was it for?

Reply:
I also did not see anything that can lead to a deadlock (hence the removal) - also, this condition could be true right after the if-block, so this is also not really protecting against anything (if a deadlock ever existed)